cirq-google/cirq_google/engine/stream_manager.py

```diff
@@ -187,25 +201,32 @@ async def _manage_stream(self) -> None:
         When the stream breaks, the stream is reopened, and all execution coroutines are notified.
 
         There is at most a single instance of this coroutine running.
+
+        Args:
+            request_queue: The queue holding requests from the execution coroutine.
         """
-        self._request_queue = asyncio.Queue()
         while True:
             try:
                 # The default gRPC client timeout is used.
                 response_iterable = await self._grpc_client.quantum_run_stream(
-                    _request_iterator(self._request_queue)
+                    _request_iterator(request_queue)
                 )
                 async for response in response_iterable:
                     self._response_demux.publish(response)
             except asyncio.CancelledError:
+                await request_queue.put(StreamManager._STOP_SIGNAL)
```

Comment:
nit: could also use `self._STOP_SIGNAL` here since `self` can also see class attributes, though personally I think using `None` is better (don't have to go look where `_STOP_SIGNAL` is defined to know what value it is). Up to you :-)

Reply:
Sg, will change it to "None" since I think it's commonly used to terminate a sequence anyway.